spylon_kernel/scala_kernel.py

```diff
@@ -41,42 +44,96 @@ class SpylonKernel(MetaKernel):
     def __init__(self, *args, **kwargs):
         self._scalamagic = None
         super(SpylonKernel, self).__init__(*args, **kwargs)
+        # Register the %%scala and %%init_spark magics
+        # The %%scala one is here only because this classes uses it
+        # to interact with the ScalaInterpreter instance
         self.register_magics(ScalaMagic)
         self.register_magics(InitSparkMagic)
         self._scalamagic = self.line_magics['scala']
 
     @property
     def scala_interpreter(self):
+        """Gets the `ScalaInterpreter` instance associated with the ScalaMagic
+        for direct use.
+
+        Returns
+        -------
+        ScalaInterpreter
+        """
         # noinspection PyProtectedMember
         intp = self._scalamagic._get_scala_interpreter()
-        assert isinstance(intp, SparkInterpreter)
+        assert isinstance(intp, ScalaInterpreter)
         return intp
 
     def get_usage(self):
+        """Gets usage information about the kernel itself.
+
+        Part of the MetaKernel API.
+        """
         return "This is spylon-kernel. It implements a Scala interpreter."
 
     def set_variable(self, name, value):
+        """Sets a variable in the kernel language.
+
+        Part of the MetaKernel API.
+
+        Parameters
+        ----------
+        name : str
+            Variable name
+        value : any
+            Variable value to set
+
+        Notes
+        -----
+        Since metakernel calls this to bind kernel into the remote space we
+        don't actually want that to happen. Simplest is just to have this
+        flag as None initially. Furthermore the metakernel will attempt to
+        set things like _i1, _i, _ii etc.  These we dont want in the kernel
+        for now.
         """
-        Set a variable in the kernel language.
-        """
-        # Since metakernel calls this to bind kernel into the remote space we
-        # don't actually want that to happen. Simplest is just to have this
-        # flag as None initially. Furthermore the metakernel will attempt to
-        # set things like _i1, _i, _ii etc.  These we dont want in the kernel
-        # for now.
         if self._scalamagic and (not name.startswith("_i")):
```

Comment:
This will happen after almost every cell execution as MetaKernel will update the FIFO queue of commands and outputs, so the logs will get pretty chatty. We could log at the debug level.